### ISSUE TITLE
Update guard() to support multiple values.

### ISSUE THREAD

**Issue by justinfagnani** [Polymer/lit-html#648]
The `guard` name has been confusing for users. `memo` seems to be common enough with it's addition to React, that it should be a bit better.

We also want to support multiple checked values.

**Comment by akc42** [comment]
Not being a react user I don't know what `memo` means, whereas `guard` I understand in the sense I know what it does - not necessarily how to use it properly (see below). I certainly view it as a sort of guarded border which prevents irrelevant changes passing through to something that is large and complex and should only be re-evaluated when necessary.

Yesterday I spent a few hours chasing down why my model had one view of the data and the display had another.  It was `guard` that was preventing a pass down of property data to an embedded element (which would have caused it to-re-render) despite (I though) being meticulous on ensuring my object reference changed when data within it had changed.  I never understood why I wasn't piercing the guard so I just I took my `guard` out and rendering worked and performance didn't seem to change.  I guess I could be guilty of too early an optimisation, but the experience has made be question the value of the directive and exactly when it should be used.  So some really good documentation would be useful.

I like the multiple values.  I ended up in one scenario just adding a field to my object which was always a  `{}` (but was easy to change to another `{}` when I wanted to pierce the guard because I really wanted to pierce the guard if one of several fields changed.
